File: examples/language-features/CSharp/dotnet/6.0/10.0/library/CSharp5/CallerInfoAttributes/CallerInfoAttributes.cs
using System.Runtime.CompilerServices;

namespace CSharpNet6_10.CSharp5.CallerInfoAttributes
{
    public class Diagnostics
    {
        // The compiler fills these optional parameters in at each call site, so
        // the caller writes nothing and there is no run-time stack inspection.
        public static string Describe(
            string message,
            [CallerMemberName] string member = "",
            [CallerFilePath] string file = "",
            [CallerLineNumber] int line = 0)
        {
            return message + " (" + member + " in " + file + ":" + line + ")";
        }

        public static string CalledFromHere()
        {
            // member becomes "CalledFromHere"; file and line describe this call.
            return Describe("checkpoint");
        }

        // An explicit argument wins over the value the compiler would inject.
        public static string CalledWithExplicitMember()
        {
            return Describe("checkpoint", "supplied-by-hand");
        }
    }
}

File: examples/language-features/VbNetNet10Latest/Baseline/CallerInfoAttributes/CallerInfoAttributes.vb
Option Strict On

Imports System.Runtime.CompilerServices

Namespace Baseline.CallerInfoAttributes
    Public Module Diagnostics
        ' The compiler fills these optional parameters in at each call site, so
        ' the caller writes nothing and no run-time stack inspection is needed.
        Public Function Describe(message As String,
                                 <CallerMemberName> Optional member As String = "",
                                 <CallerFilePath> Optional file As String = "",
                                 <CallerLineNumber> Optional line As Integer = 0) As String
            Return message & " (" & member & " in " & file & ":" & line.ToString() & ")"
        End Function

        Public Function CalledFromHere() As String
            ' member becomes "CalledFromHere"; file and line describe this call.
            Return Describe("checkpoint")
        End Function

        ' An explicit argument wins over the value the compiler would inject.
        Public Function CalledWithExplicitMember() As String
            Return Describe("checkpoint", "supplied-by-hand")
        End Function
    End Module
End Namespace
